tests/ui/cypress/specs/multi_user_openid_connect.js

```diff
@@ -0,0 +1,289 @@
+    // and then clicking 'Ok'. This should fail, since the token can't be refereshed.
+    cy.intercept('POST', '/api/v1/cas').as('createCA')
+    cy.get('main input').type('some-handle-name')
+    cy.get('main button').click()
+    cy.get('.el-message-box__btns button:nth-child(2)').click()
+    cy.wait('@createCA').its('response.statusCode').should('eq', 401)
+
+    // verify that we are shown the OpenID Connect provider login page
+    // cy.intercept('GET', '/api/v1/authorized').as('isAuthorized')
+    cy.intercept('GET', '/auth/login').as('getLoginURL')
+    cy.intercept('GET', /^http:\/\/localhost:1818\/authorize.+/).as('oidcLoginForm')
+
+    cy.visit('/')
+    // not sure why but even though the 401 response is sent and Cypress debug
+    // logs show it, the following test never finds a match...
+    // cy.wait('@isAuthorized').its('response.statusCode').should('eq', 401)
+    cy.wait(['@getLoginURL', '@oidcLoginForm'])
+
+    cy.url().should('not.include', Cypress.config('baseUrl'))
+    cy.contains('Mock OpenID Connect login form')
+    cy.get('input[name="username"]')
+  })
+
+  it('[valid refresh] Login with short-live non-refreshable token and try to create a CA', () => {
+    cy.intercept('GET', '/api/v1/authorized').as('isAuthorized')
+    cy.visit('/')
+
+    cy.wait('@isAuthorized').its('response.statusCode').should('eq', 403)
+    cy.url().should('not.include', Cypress.config('baseUrl'))
+    cy.contains('Mock OpenID Connect login form')
+    cy.get('input[name="username"]').clear().type(shorttoken.u)
+
+    cy.intercept('GET', '/index.html').as('postLoginIndexFetch')
+    cy.intercept('GET', '/api/v1/authorized').as('isAuthorized')
+    cy.contains('Sign In').click()
+
+    cy.wait('@postLoginIndexFetch').its('response.statusCode').should('eq', 200)
+    cy.wait('@isAuthorized').its('response.statusCode').should('eq', 200)
+    cy.url().should('include', Cypress.config('baseUrl'))
+    cy.contains('Sign In').should('not.exist')
+    cy.get('#userinfo').click()
+    cy.get('#userinfo_table').contains(shorttoken.u)
+    cy.contains(shorttoken.u)
+    cy.contains('Welcome to Krill')
+
+    // the token has a lifetime of 5 second and no refresh token
+    // wait 6 seconds...
+    // note: a shorter token with a 1 second lifetime doesn't work in the GitHub
+    // Action runner environment because the token has sometimes already expired
+    // by the time Krill verifies it!
+    cy.wait(6000)
+
+    // Try to create a CA, by typing in the input, clicking the 'Create CA' button
+    // and then clicking 'Ok'. This should fail, since the token can't be refreshed.
+    cy.intercept('POST', '/api/v1/cas').as('createCA')
+    cy.get('main input').type('some-handle-name')
+    cy.get('main button').click()
+    cy.get('.el-message-box__btns button:nth-child(2)').click()
+    cy.wait('@createCA').its('response.statusCode').should('eq', 401)
+  })
+  ;[...create_ca_settings_401, ...create_ca_settings_403].forEach((t) =>
```

Comment:
Interesting, prettier (the JS formatter) really, really wants this if you disable semis.